### DISCUSSION TITLE
html-audio-response: WebM vs other file formats (e.g. FLAC)

### DISCUSSION THREAD

**Discussion by ghost** [jspsych/jsPsych#2534] [Q&A] [answered]
Hi! The new html-audio-response plugin is fantastic. I wanted to learn a bit more about the format of the audio after collection. 

Bear with my limited knowledge here:
I believe it is stored as a base64 string representation of a WebM audio file. This means the compressed audio is lossy, correct? In my lab, we use the FLAC format for lossless compression- is there a reason to use WebM over other formats like FLAC, and is it possible to modify the plugin so that the audio is saved as base64 string representing a FLAC format?

Thanks!

**Comment by ghost** [comment]
Hello! Bumping this just in case anyone has any thoughts as to using WebM vs other formats for collecting audio response. Thanks!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @HappyRhino12, great question. The webm format is definitely not ideal. Unfortunately I'm not aware of any other way to store browser-based audio recordings via the MediaRecorder API. If anyone else knows how to do this, I'd be very interested to hear about it so that we can modify the plugin.

And just to be clear - it's definitely possible to convert the webm data into another audio file format _after_ the data has been collected. (I do this offline using audio conversion software like ffmpeg or Kabuu). This is helpful for converting the files into a more common file format, which can then be read by most audio players etc. But this doesn't help with the problem of initial lossy vs lossless compression of the recordings.
